hack/dev-env/create-agent-config.sh

```diff
@@ -0,0 +1,54 @@
+#!/bin/bash
+
+# Copyright 2025 The argocd-agent Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -eo pipefail
+
+RECREATE="$1"
+
+SCRIPTPATH="$( cd -- "$(dirname "$0")" >/dev/null 2>&1 ; pwd -P )"
+BASEPATH="$( cd -- "$(dirname "$0")/../.." >/dev/null 2>&1 ; pwd -P )"
+AGENTCTL=${BASEPATH}/dist/argocd-agentctl
+export ARGOCD_AGENT_CONTEXT=vcluster-control-plane
+
+IPADDR=$(ip r show default | sed -e 's,.*\ src\ ,,' | sed -e 's,\ metric.*$,,')
```

Comment:
This will be incompatible with MacOS environments. I do not have access to a Mac, so I'm not sure how to come up with a portable solution.